src/main/java/picard/sam/markduplicates/UmiAwareDuplicateSetIterator.java

```diff
@@ -88,7 +90,11 @@
     public void close() {
         isOpen = false;
         wrappedIterator.close();
-        metrics.calculateDerivedFields();
+
+        // Calculate derived fields for UMI metrics over each library
+        for (final Map.Entry<String, UmiMetrics> metric : umiMetricsMap.entrySet()) {
+           metric.getValue().calculateDerivedFields();
```

Comment:
It looks like this can be replaced with a loop over the map's values, since the key is unused.
```
for (UmiMetrics metric : umiMetricsMap.values()) {
    metric.calculateDerivedFields();
}
```